### PR TITLE
Reactivation PC-Met access

### PR DESCRIPTION
**Pull request by August2111**
Brief summary of the changes
----------------------------

More then 2-3 years ago Flugwetter.de changed a lot of URI fields for accessing the DWD content like Radar, Sat images and so on. Unfortunately there was no maintainance of this component since more then 4 years. 
In this PR I changed all related things, that this access and all the views are functional. For example on GDI this couldn't display anything, because without gdiplus (or maybe an other lib) there is no possibiltiy to read JPEG files (anf PNG, GIF,... and so on) 

Related issues and discussions
------------------------------
With this functional component there are a lot of things possible to made:
* automatic update of the last current sat image
* a view with zoom in, zoom out and move... (the complete picture doesn't have the best resolution...)
* a (short) video of last pictures

At least this feature is functional again...

### CODE REVIEW

File: src/Dialogs/Airspace/dlgAirspaceWarnings.cpp [hunk 1]
```diff
@@ -330,6 +330,7 @@ AirspaceWarningListWidget::OnPaintItem(Canvas &canvas,
   auto [text_rc, altitude_rc] =
     text_altitude_rc.VerticalSplit(text_altitude_rc.right - (padding + altitude_width));
   text_rc.right -= padding;
+  altitude_rc.bottom = altitude_rc.bottom;  // unused variable?
```
Comment (MaxKellermann) [inline src/Dialogs/Airspace/dlgAirspaceWarnings.cpp:333]: Doesn't belong in this commit

File: src/Weather/PCMet/Images.cpp [hunk 2]
```diff
@@ -101,8 +111,7 @@ const PCMet::ImageType PCMet::image_types[] = {
 Bitmap
 PCMet::DownloadLatestImage(const char *type, const char *area,
                            const PCMetSettings &settings,
-                           JobRunner &runner)
-{
+                           JobRunner &runner) {
```
Comment (MaxKellermann) [inline src/Weather/PCMet/Images.cpp]: Doesn't belong in this commit

File: src/Weather/PCMet/Images.cpp [hunk 3]
```diff
@@ -154,7 +163,6 @@ PCMet::DownloadLatestImage(const char *type, const char *area,
     if (!runner.Run(job2))
       return Bitmap();
   }
-
```
Comment (MaxKellermann) [inline src/Weather/PCMet/Images.cpp:157]: Why?

File: src/ui/canvas/Bitmap.hpp [hunk 4]
```diff
@@ -106,7 +108,8 @@ class Bitmap final
 #elif defined(USE_MEMORY_CANVAS)
   WritableImageBuffer<BitmapPixelTraits> buffer = WritableImageBuffer<BitmapPixelTraits>::Empty();
 #else
-  HBITMAP bitmap = nullptr;
+  static std::unordered_map<HBITMAP, size_t> bitmap_map;
```
Comment (MaxKellermann) [inline src/ui/canvas/Bitmap.hpp:111]: Hidden global variable, and it's not thread-safe. This will crash XCSoar.

File: src/ui/canvas/gdi/Bitmap.cpp [hunk 5]
```diff
@@ -22,36 +22,64 @@ Copyright_License {
...
+
 }
 
 bool
 Bitmap::LoadFile(Path path)
 {
-  return false;
+  hbitmap = GdiLoadImage(path.c_str());
+  if (!IsDefined()) {
+    LogFormat(_T("Bitmap couldn't load:  %d!"), GetLastError());
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/Bitmap.cpp:54]: Throw exception instead of logging

File: src/Startup.cpp [hunk 6]
```diff
@@ -140,13 +144,14 @@ AfterStartup()
   } catch (...) {
       LogError(std::current_exception());
   }
-
+  LogFormat("next: processGlideComputer");
```
Comment (MaxKellermann) [inline src/Startup.cpp:147]: Useless log message

File: src/Startup.cpp [hunk 7]
```diff
@@ -140,13 +144,14 @@ AfterStartup()
...
   if (is_simulator()) {
     InputEvents::processGlideComputer(GCE_STARTUP_SIMULATOR);
   } else {
     InputEvents::processGlideComputer(GCE_STARTUP_REAL);
   }
 
+  LogFormat("next: LoadDefaultTask");
```
Comment (MaxKellermann) [inline src/Startup.cpp:154]: Useless log message

File: src/Startup.cpp [hunk 8]
```diff
@@ -159,11 +164,15 @@ AfterStartup()
     protected_task_manager->TaskCommit(*defaultTask);
   }
 
+  LogFormat("next: TaskManager->Resume");
```
Comment (MaxKellermann) [inline src/Startup.cpp:167]: Useless log message

File: src/Startup.cpp [hunk 9]
```diff
@@ -159,11 +164,15 @@ AfterStartup()
...
   task_manager->Resume();
 
+  LogFormat("next: InfoBoxManager::SetDirty()");
```
Comment (MaxKellermann) [inline src/Startup.cpp:170]: Useless log message

File: src/Startup.cpp [hunk 10]
```diff
@@ -159,11 +164,15 @@ AfterStartup()
...
   InfoBoxManager::SetDirty();
 
+  LogFormat("next: ForceCalculation");
```
Comment (MaxKellermann) [inline src/Startup.cpp:173]: Useless log message

File: src/Startup.cpp [hunk 11]
```diff
@@ -159,11 +164,15 @@ AfterStartup()
...
   ForceCalculation();
+  LogFormat("end: AfterStartup");
```
Comment (MaxKellermann) [inline src/Startup.cpp:175]: Useless log message

File: src/Startup.cpp [hunk 12]
```diff
@@ -465,7 +478,8 @@ Startup()
   operation.Hide();
 
   main_window->FinishStartup();
-
+  
+  LogFormat("end: Startup");
```
Comment (MaxKellermann) [inline src/Startup.cpp:482]: Useless log message

File: src/ui/canvas/gdi/Bitmap.cpp [hunk 13]
```diff
@@ -26,7 +26,9 @@ Copyright_License {
 
 #include "Screen/Debug.hpp"
 #include "system/Path.hpp"
+#include "LogFile.hpp"
 
+// #include <cassert>
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/Bitmap.cpp]: Why?

File: src/ui/canvas/gdi/Bitmap.cpp [hunk 14]
```diff
@@ -26,7 +26,9 @@ Copyright_License {
 
 #include "Screen/Debug.hpp"
 #include "system/Path.hpp"
+#include "LogFile.hpp"
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/Bitmap.cpp:29]: Why?

File: src/ui/canvas/gdi/GdiPlusBitmap.cpp [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+// can load: BMP, GIF, JPEG, PNG, TIFF, Exif, WMF, and EMF
+HBITMAP GdiLoadImage(const TCHAR* filename)
+{
+  HBITMAP result =nullptr;
+  Bitmap bitmap(filename, false);
+  bitmap.GetHBITMAP(0, &result);
+  return result;
+}
+
+#if 0
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/GdiPlusBitmap.cpp:46]: Huh?

File: src/ui/canvas/gdi/GdiPlusBitmap.hpp [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+// #define min std::min
+// #define max std::max
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/GdiPlusBitmap.hpp:24]: Huh?

File: src/ui/canvas/gdi/GdiPlusBitmap.hpp [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+#include <windows.h>
+// namespace Gdi {
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/GdiPlusBitmap.hpp:26]: Huh?

File: src/net/http/Request.cpp [hunk 18]
```diff
@@ -41,6 +41,11 @@ Net::Request::Request(Session &_session, ResponseHandler &_handler,
   handle.SetUserAgent(user_agent);
   handle.SetFailOnError(true);
   handle.SetWriteFunction(WriteCallback, this);
+#if defined(_WIN32)
+  // otherwise under Windows it is no access to https urls per SSL
+  // maybe the better option is to adjust the openssl component!
```
Comment (MaxKellermann) [inline src/net/http/Request.cpp:46]: No, we should fix the underlying issue instead of doing such a kludge.

**Review by MaxKellermann — CHANGES_REQUESTED** [review]

File: src/Dialogs/Airspace/dlgAirspaceWarnings.cpp (hunk 1 above)
Comment (August2111) [reply src/Dialogs/Airspace/dlgAirspaceWarnings.cpp:333]: Ok, I would move it in an extra commit. Is it possible to do this in this PR?

File: src/Weather/PCMet/Images.cpp (hunk 2 above)
Comment (August2111) [reply src/Weather/PCMet/Images.cpp]: I will remove this!

File: src/ui/canvas/Bitmap.hpp (hunk 4 above)
Comment (August2111) [reply src/ui/canvas/Bitmap.hpp:111]: This I don't like
What do you prefere? An use counter? I don't like the member setting to null in the source at a object copy, see line 32
Bitmap::Bitmap(Bitmap &&src)
  :bitmap(src.bitmap)
{
  src.bitmap = nullptr;
}
This means, after a copy you cannot access to the source object?

File: src/ui/canvas/gdi/Bitmap.cpp (hunk 14 above)
Comment (August2111) [reply src/ui/canvas/gdi/Bitmap.cpp:29]: Because LogFormat lines, - if they not necessary, this can be removed!

File: src/ui/canvas/gdi/GdiPlusBitmap.cpp (hunk 15 above)
Comment (August2111) [reply src/ui/canvas/gdi/GdiPlusBitmap.cpp:46]: :( my mistake

File: src/ui/canvas/gdi/GdiPlusBitmap.hpp (hunk 17 above)
Comment (August2111) [reply src/ui/canvas/gdi/GdiPlusBitmap.hpp:26]: :( my mistake too

File: src/net/http/Request.cpp (hunk 18 above)
Comment (August2111) [reply src/net/http/Request.cpp:46]: This needs much more investigations, I thought, a workaround is better as a complete failure: In the moment under Windows XCSoar has NO access to any https page yet

**Comment by August2111** [comment]
I will resolve the issues and I want to make an update of the PR as soon as possible!

File: src/ui/canvas/Bitmap.hpp (hunk 4 above)
Comment (MaxKellermann) [reply src/ui/canvas/Bitmap.hpp]: This is not a copy constructor, it's a move constructor, and it's supposed to make the moved-from object invalid.
Why a counter?

File: src/Dialogs/Airspace/dlgAirspaceWarnings.cpp (hunk 1 above)
Comment (MaxKellermann) [reply src/Dialogs/Airspace/dlgAirspaceWarnings.cpp]: Yes.

File: src/Weather/PCMet/Images.cpp [hunk 19]
```diff
@@ -99,10 +109,11 @@ const PCMet::ImageType PCMet::image_types[] = {
 };
 
 Bitmap
-PCMet::DownloadLatestImage(const char *type, const char *area,
-                           const PCMetSettings &settings,
-                           JobRunner &runner)
+PCMet::DownloadLatestImage(const char* type, const char* area,
+  const PCMetSettings& settings,
+  JobRunner& runner)
```
Comment (MaxKellermann) [inline src/Weather/PCMet/Images.cpp:114]: Don't reformat code, unless you submit a commit which does *only* that.

File: src/Weather/PCMet/Images.cpp [hunk 20]
```diff
@@ -99,10 +109,11 @@ const PCMet::ImageType PCMet::image_types[] = {
...
 {
+  Bitmap bitmap;
```
Comment (MaxKellermann) [inline src/Weather/PCMet/Images.cpp:116]: Why?

File: src/Weather/PCMet/Images.cpp [hunk 21]
```diff
@@ -118,44 +129,46 @@ PCMet::DownloadLatestImage(const char *type, const char *area,
...
-  char *img = strstr(buffer, "<img name=\"bild\" src=\"/");
+  char* img = strstr(buffer, "<img name=\"bild\" src=\"/");
   if (img == nullptr)
-    return Bitmap();
+    return bitmap;
 
-  char *src = img + sizeof(img_needle) - 2;
-  char *end = strchr(src + 1, '"');
+  char* src = img + sizeof(img_needle) - 2;
+  char* end = strchr(src + 1, '"');
```
Comment (MaxKellermann) [inline src/Weather/PCMet/Images.cpp:142]: More code reformatting

File: src/Dialogs/Weather/PCMetDialog.cpp [hunk 22]
```diff
@@ -157,6 +156,7 @@ class ImageTypeListWidget final : public TextListWidget {
   }
 };
 
+//-----------------------------------------------------------------------------
```
Comment (MaxKellermann) [inline src/Dialogs/Weather/PCMetDialog.cpp:159]: ??

File: src/Dialogs/Weather/PCMetDialog.cpp [hunk 23]
```diff
@@ -44,11 +44,10 @@ Copyright_License {
...
 {
-  TWidgetDialog<ViewImageWidget>
-    dialog(WidgetDialog::Full{}, UIGlobals::GetMainWindow(),
-           UIGlobals::GetDialogLook(),
-           _T("pc_met"));
-  dialog.SetWidget();
+  ViewImageWidget widget(bitmap);
+  WidgetDialog dialog(WidgetDialog::Full{}, UIGlobals::GetMainWindow(),
+                      UIGlobals::GetDialogLook(),
+                      _T("pc_met"), &widget);
```
Comment (MaxKellermann) [inline src/Dialogs/Weather/PCMetDialog.cpp:50]: Why? Your commit message doesn't explain.

File: src/Dialogs/Airspace/dlgAirspaceWarnings.cpp [hunk 24]
```diff
@@ -330,6 +330,7 @@ AirspaceWarningListWidget::OnPaintItem(Canvas &canvas,
   auto [text_rc, altitude_rc] =
     text_altitude_rc.VerticalSplit(text_altitude_rc.right - (padding + altitude_width));
   text_rc.right -= padding;
+  altitude_rc.bottom = altitude_rc.bottom;  // unused variable
```
Comment (MaxKellermann) [inline src/Dialogs/Airspace/dlgAirspaceWarnings.cpp:333]: This isn't an unused variable. `altitude_rc` is a variable, but `altitude_rc.bottom` is not.

Comment (August2111) [reply src/Dialogs/Airspace/dlgAirspaceWarnings.cpp:333]: My compiler says that altitude_rc is an unused variable, and so i made this fix ... :-( 
Or should I only change the comment?

Comment (MaxKellermann) [reply src/Dialogs/Airspace/dlgAirspaceWarnings.cpp:333]: See, it's about `altitude_rc`, not about `altitude_rc.bottom`.
If the variable is unused, remove it instead of applying a workaround.

**Comment by August2111** [comment]
Excuse me I'm not complete ready yet with my code - tomorrow I want to make it really clean, f.e. also test the previous dialog with  TWidgetDialog too... Maybe I copied this from an other state or file ...
I want to compare every line with the repo - and than I hope you are happy with .

The part of curl I want to exclude from this PR: This has nothing to do with PC-Met - and have an impact of the access of all https sites under Windows. 

I have a solution for this - not a kludge - but probably the curl component itself needs an other building setup on mingw ;-(
In the meanwhile with MSVC I have the communication to SSL-Sites running WITH PeerVerification, but the setup for mingw isn't ready yet so far. Hopefully on monday latest I will have start a PR with this!

File: src/ui/canvas/Bitmap.hpp (hunk 4 above)
Comment (August2111) [reply src/ui/canvas/Bitmap.hpp]: Unfortunately I didn't know the move constructor ;-( I learned, this is defined in C++11. With such a functional possibility their is no ref counter necessary, of course...

File: src/Dialogs/Weather/PCMetDialog.cpp (hunk 23 above)
Comment (August2111) [reply src/Dialogs/Weather/PCMetDialog.cpp]: Changed back

File: src/Weather/PCMet/Images.cpp (hunk 21 above)
Comment (August2111) [reply src/Weather/PCMet/Images.cpp]: I have to avoid the changes by my editor!

File: src/Dialogs/Airspace/dlgAirspaceWarnings.cpp (hunk 24 above)
Comment (August2111) [reply src/Dialogs/Airspace/dlgAirspaceWarnings.cpp]: Done

**Comment by August2111** [comment]
Now I'm ready with my review yet!
Hopefully it is ok now.
Thanks for your hints, Max

File: src/Weather/PCMet/Images.cpp (hunk 3 above)
Comment (MaxKellermann) [reply src/Weather/PCMet/Images.cpp:157]: This can crash XCSoar due to TOCTOU bug.

File: src/Weather/PCMet/Images.cpp (hunk 20 above)
Comment (MaxKellermann) [reply src/Weather/PCMet/Images.cpp:116]: > Creates Bitmap() once - and return with this instead of creation on every return place

Again: why? This whole commit seems rather pointless.

File: src/ui/canvas/gdi/GdiPlusBitmap.hpp [hunk 25]
```diff
@@ -0,0 +1,27 @@
...
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/GdiPlusBitmap.hpp:23]: Include guard is missing

File: src/ui/canvas/gdi/GdiPlusBitmap.hpp [hunk 26]
```diff
@@ -0,0 +1,27 @@
...
+#include <windows.h>
+extern HBITMAP GdiLoadImage(const TCHAR* filename);
+extern void GdiStartup();
+extern void GdiShutdown();
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/GdiPlusBitmap.hpp:27]: `extern` not necessary in C++

File: src/Startup.cpp [hunk 27]
```diff
@@ -457,6 +461,10 @@ Startup()
 #endif
 #endif
 
+#ifdef USE_GDI
+  GdiStartup();
```
Comment (MaxKellermann) [inline src/Startup.cpp:465]: This needs to be called in all programs using the UI toolkit library - i.e. all those programs in `test/src/`. You need a better approach for this, e.g. in `class ScreenGlobalInit` which is used by all relevant programs.

File: src/ui/canvas/gdi/GdiPlusBitmap.cpp [hunk 28]
```diff
@@ -0,0 +1,58 @@
...
+}
+*/
+
+#include "GdiPlusBitmap.hpp"
+
+#if defined(_MSC_VER)
+# include <algorithm>
+# define min std::min
+# define max std::max
+#endif  // _MSC_VER
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/GdiPlusBitmap.cpp:30]: Why?

File: src/ui/canvas/gdi/GdiPlusBitmap.cpp [hunk 29]
```diff
@@ -0,0 +1,58 @@
...
+#include <gdiplus.h>
+
+using namespace Gdiplus;
+
+// can load: BMP, GIF, JPEG, PNG, TIFF, Exif, WMF, and EMF
+HBITMAP GdiLoadImage(const TCHAR* filename)
+{
+  HBITMAP result =nullptr;
+  Bitmap bitmap(filename, false);
+  bitmap.GetHBITMAP(0, &result);
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/GdiPlusBitmap.cpp:42]: Does this even compile?
https://docs.microsoft.com/en-us/windows/win32/api/gdiplusheaders/nf-gdiplusheaders-bitmap-gethbitmap
> Status GetHBITMAP(
>  const Color & colorBackground,
>  HBITMAP       *hbmReturn
>);

How can you pass `0` for the `const Color&`?

File: src/ui/canvas/gdi/GdiPlusBitmap.cpp [hunk 30]
```diff
@@ -0,0 +1,58 @@
...
+#include <unknwn.h>
+#include <gdiplus.h>
+
+using namespace Gdiplus;
+
+// can load: BMP, GIF, JPEG, PNG, TIFF, Exif, WMF, and EMF
+HBITMAP GdiLoadImage(const TCHAR* filename)
+{
+  HBITMAP result =nullptr;
+  Bitmap bitmap(filename, false);
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/GdiPlusBitmap.cpp:41]: Which constructor is that? I can't find it at https://docs.microsoft.com/en-us/windows/win32/api/gdiplusheaders/nl-gdiplusheaders-bitmap

File: src/ui/canvas/gdi/Bitmap.cpp [hunk 31]
```diff
@@ -45,10 +51,7 @@ Bitmap::Reset()
   if (bitmap != nullptr) {
     assert(IsScreenInitialized());
 
-#ifndef NDEBUG
-    bool success =
-#endif
-      ::DeleteObject(bitmap);
+    bool success = ::DeleteObject(bitmap);
```
Comment (MaxKellermann) [inline src/ui/canvas/gdi/Bitmap.cpp:54]: Why?

**Review by MaxKellermann — CHANGES_REQUESTED** [review]

File: src/ui/canvas/gdi/GdiPlusBitmap.hpp (hunk 26 above)
Comment (August2111) [reply src/ui/canvas/gdi/GdiPlusBitmap.hpp:27]: I know, but for me it looks better...
I will remove it!

File: src/ui/canvas/gdi/GdiPlusBitmap.hpp (hunk 25 above)
Comment (August2111) [reply src/ui/canvas/gdi/GdiPlusBitmap.hpp:23]: Ok, of course...

File: src/ui/canvas/gdi/Bitmap.cpp (hunk 31 above)
Comment (August2111) [reply src/ui/canvas/gdi/Bitmap.cpp:54]: With a defined NDEBUG you have an assert test with an undefined variable in the next line:
assert(success)
This makes no sense for me - and the macro switch has to be removed - or the assert have to chambered with  the same definition!

Comment (MaxKellermann) [reply src/ui/canvas/gdi/Bitmap.cpp:54]: With `NDEBUG`, all `assert()` calls are removed.

Comment (August2111) [reply src/ui/canvas/gdi/Bitmap.cpp:54]: Ok, I found this in the assert.h yet. 
So 'success' is an unused variable without NDEBUG ;-( 
Without this backround knowledge it is diffecult to read (and understand).
As I said: I'm very surprised how much new things I have to learn, before I can follow your style and intentions completely.

Comment (MaxKellermann) [reply src/ui/canvas/gdi/Bitmap.cpp:54]: `assert()` exists since at least C89, i.e. 32 years ago.

**Comment by MaxKellermann** [comment]
Please rebase before you push again to your PR branch; I have already cherry-picked several of your commits, and rebasing will drop them automatically from the PR branch.

File: src/Weather/PCMet/Images.cpp (hunk 3 above)
Comment (MaxKellermann) [reply src/Weather/PCMet/Images.cpp:157]: Still problem still exists

**Comment by August2111** [comment]
How can I mark the 'requested changes' as ready? Only with the push to my origin? Or it is possible to block your visiting by another way? Before you make a deeper look inside I want to do a review by myself - but you was faster the me...
Before I would set a 'ready' flag I want to do the rebasing :) (Also a workflow that is complete new for me)

File: src/Weather/PCMet/Images.cpp (hunk 3 above)
Comment (August2111) [reply src/Weather/PCMet/Images.cpp:157]: I didn't know anything about this bug - I didn't changed here the old code. But I will look for this Bug - and hopefully I can solve!

Comment (MaxKellermann) [reply src/Weather/PCMet/Images.cpp:157]: You added an `assert()`.

Comment (August2111) [reply src/Weather/PCMet/Images.cpp:157]: 'Time-of-Check-to-Time-of-Use-Problem' - ok, I will check it

**Comment by August2111** [comment]
The last push was wrong, I have to try to setup it again ;-(

File: src/Weather/PCMet/Images.cpp (hunk 3 above)
Comment (August2111) [reply src/Weather/PCMet/Images.cpp:157]: Ok, I changed it. But why? I had learned that an assert is operative only in Debug mode (With NDEBUG its only 0)?

File: src/ui/canvas/gdi/GdiPlusBitmap.cpp (hunk 28 above)
Comment (August2111) [reply src/ui/canvas/gdi/GdiPlusBitmap.cpp]: Inside the gdiplus headers the 'min' and 'max' are using very often, but with the define NOMINMAX they are not available. So you have define it explicitly. The recommandations found in stackoverflow (https://stackoverflow.com/questions/4913922/possible-problems-with-nominmax-on-visual-c) shows exactly this problem, but they are using the 'using' directive, not the #define - and so I changed it in this way.

**Comment by August2111** [comment]
From my point of view I have done it - hopefully in the correct way...
I had also prepared a PR with a change of the curl component - but in your last commits I found the whole changes that I would proposed... It's poosible to use the openssl but here you need an additional certification file - and this also like a kludge for me. The use uf SCHANNEL ( = the native WinSSL) looks for me much better :) So I can delete my 'curl' branch - I think all is fine yet!

File: src/Weather/PCMet/Images.cpp (hunk 3 above)
Comment (MaxKellermann) [reply src/Weather/PCMet/Images.cpp:157]: True, and it will crash only debug builds, but it's still a debug.

**Comment by MaxKellermann** [comment]
I merged this manually, with several bug fixes.

File: src/Startup.cpp (hunk 27 above)
Comment (August2111) [reply src/Startup.cpp:465]: Ok, I forgot this move to  class ScreenGlobalInit  - your way is much better!
